Rust/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+mod boundingbox;
+mod cut;
+mod imputevisitor;
+mod intervalstoremanager;
+pub mod multidimdatawithkey;
+mod nodestore;
+mod nodeview;
+mod pointstore;
+mod randomcuttree;
+pub mod rcf;
+mod sampler;
+mod samplerplustree;
+pub mod samplesummary;
+mod scalarscorevisitor;
+mod types;
+mod visitor;
+mod conditionalfieldsummarizer;
+extern crate rand;
+
+use num::abs;
+use crate::rcf::{create_rcf, RCF};
+extern crate rand_chacha;
+
+pub fn L1distance(a: &[f32], b : &[f32]) -> f64{
```

Comment:
In all the distance calculations in this file, would it make sense to convert numbers to f64 before performing subtractions to avoid underflow?

Reply:
Good callout. Fixing.